processor/src/host/mod.rs

```diff
@@ -91,20 +119,30 @@ pub trait AsyncHost: BaseHost {
     // REQUIRED METHODS
     // --------------------------------------------------------------------------------------------
 
-    // Note: we don't use the `async` keyword in both of these methods, since we need to specify the
-    // `+ Send` bound to the returned Future, and `async` doesn't allow us to do that.
+    // Note: we don't use the `async` keyword in this method, since we need to specify the `+ Send`
+    // bound to the returned Future, and `async` doesn't allow us to do that.
 
-    /// Returns MAST forest corresponding to the specified digest, or None if the MAST forest for
-    /// this digest could not be found in this [AsyncHost].
-    fn get_mast_forest(
-        &self,
-        node_digest: &Word,
-    ) -> impl Future<Output = Option<Arc<MastForest>>> + Send;
-
-    /// Handles the event emitted from the VM.
+    /// Handles the event emitted from the VM and provides advice mutations to be applied to
+    /// the advice provider.
     fn on_event(
         &mut self,
-        process: &mut ProcessState<'_>,
+        process: &ProcessState<'_>,
         event_id: u32,
-    ) -> impl Future<Output = Result<(), EventError>> + Send;
+    ) -> impl FutureMaybeSend<Result<Vec<AdviceMutation>, EventError>>;
 }
+
+/// Alias for a `Future`
+///
+/// If feature `std` is enabled, we add `Send` to the required bounds, otherwise we do not. This
```

Comment:
this still mentions the `std` feature - should be updated to `wasm` target family.

Should we also mention that this is automatically implemented for all `Future`s, similar to how the standard library [does for `Into`](https://doc.rust-lang.org/std/convert/trait.Into.html) (with respect to the comment about a blanket implementation for all types that implement `From`)?

Reply:
done!